Dockerfile

```diff
@@ -32,16 +47,22 @@ COPY src/Pipfile src/Pipfile.lock ./
 # VIRTUAL_ENV environment variable if it is set.
 RUN pipenv sync --clear --verbose
 
-FROM python:3.10.7-alpine3.16 AS build-stage
+# The version of Python used here should match the version of the Alpine
+# python3 package installed in the compile-stage.
+FROM python:3.10.10-alpine3.17 AS build-stage
```

Comment:
Is there any way to capture the Python version live, from the compile stage, put it into a variable, and then reference it in the `FROM`?

Reply:
Why is it fragile and error prone?  Your comment states that "[t]he version of Python used here should match the version of the Alpine `python3` package package installed in the compile-stage".  Capturing the Python version live from the compile stage and then using that value to define the base image to use for the build stage would seem to be _less_ fragile, since it wouldn't rely on a human to manually check the Python version in the compile stage and verify that it matches the version in the base image used in the build stage.

Reply:
It would be fragile because any change to the Python version string output would break the functionality. After trying I don't see a way to extract the Python version from the first base image to use in the second base image's tag due to when the `FROM` instructions are evaluated.

Reply:
> After trying I don't see a way to extract the Python version from the first base image to use in the second base image's tag due to when the `FROM` instructions are evaluated.

Are you saying that the `FROM` instructions are not evaluated sequentially with the other `Dockerfile` directives, but are instead all evaluated in one fell swoop at the beginning of the `docker build`?

Reply:
That is what appears to be the case. Also `ARG` and `ENV` values are scopes to the specific build stage they are defined in with the only exception being `ARG` values declared before the first `FROM` instruction.